### PR TITLE
Add tests for palette mode

### PR DESCRIPTION
**Pull request by Cidana-Developers**
1. add unit test for count color.
- eb_av1_count_colors
- av1_count_colors_highbd
2. add unit test for kmeans.
- av1_k_means_dim1
- av1_k_means_dim2
3. add api parameter test for palette mode
4. add e2e parameter test for palette mode

### CODE REVIEW

File: test/api_test/params.h [hunk 1]
```diff
@@ -591,6 +591,21 @@ static const vector<uint32_t> invalid_search_area_height = {
...
+/* Palette Mode
+ *-1:Auto Mode(ON at level6 when SC is detected)
+ * 0:OFF
+ * 1:Slow    NIC=7/4/4
+ * 2:        NIC=7/2/2
+ * 3:        NIC=7/2/2 + No K means for non ref
+ * 4:        NIC=4/2/1
+ * 5:        NIC=4/2/1 + No K means for Inter frame
+ * 6:Fastest NIC=4/2/1 + No K means for non base + step for non base for
+ * most dominent
```
Comment (andrey-norkin) [inline test/api_test/params.h]: spelling: dominant

Comment (Cidana-Developers) [reply test/api_test/params.h]: thank you for remanding, it is fixing

File: test/PaletteModeUtilTest.cc [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        uint8_t tmp[MAX_PALETTE_SQUARE] = {0};
+        for (size_t i = 0; i < MAX_PALETTE_SQUARE; i++)
+            data_[i] = tmp[i] = palette[rnd_.random() % max_colors];
+        delete[] palette;
+        int val_count[MAX_PALETTE_SQUARE] = {0};
+        return eb_av1_count_colors(tmp, 64, 64, 64, val_count);
+    }
+
+    void run_test(size_t times) {
+        uint8_t indeces[MAX_PALETTE_SQUARE] = {0};
```
[the remark this answers is not in the document]
Comment (Cidana-Developers) [reply test/PaletteModeUtilTest.cc]: thank you for remanding, it is fixing

**Comment by hassount** [comment]
@andrey-norkin 

Items resolved, can we merge this ?

**Pull request merged**